lib/src/has_next_iterator.dart

```diff
@@ -2,80 +2,25 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-///
-/// Re-implementation of [Iterator] which saves the most recent result from
-/// [moveNext].
-///
-class HasNextIterator<E> {
-  final List<E> _items;
-  final int _length;
-  bool _hasNext = true;
-  int _index = 0;
-  E? _current;
+/// [Iterator] which wraps another [Iterator] and saves the most recent result
```

Comment:
@natebosch how do you feel about this?

The way the iterator is used in this case makes it very difficult to track outside the iterator class.